x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/reducer.ts

```diff
@@ -132,6 +165,8 @@ export const hostListReducer: ImmutableReducer<HostState, AppAction> = (
       error: undefined,
       detailsError: undefined,
       policyResponseError: undefined,
+      loading: true,
```

Comment:
I'm actually confused by this return here (unrelated to your change), which may be causing the state to be updated on every URL change.

I think the return here should be to reset the state for hosts if the user is leaving the Endpoints page (list and details) and if not leaving, then do nothing (just return `state`).

so maybe something like:

```typescript
if (
  (wasPreviouslyOnListPage || wasPreviouslyOnDetailsPage) &&
  !isCurrentlyOnListPage && 
  !iscurrentlyOnDetailsPage
) {
  return initialHostListState();
}

return state;
```

Reply:
I'm actually removing this as it's what's causing the test failures